In [1]:
import numpy as np
import pandas as pd
from scipy import stats


In [2]:
education_districtwise = pd.read_csv('/kaggle/input/education-districtwise-file/education_districtwise.csv')


In [3]:
sampled_data = education_districtwise.sample(n=50, replace=True, random_state=31208)
sampled_data

DISTNAME STATNAME  BLOCKS  VILLAGES  CLUSTERS  TOTPOPULAT  OVERALL_LI
615  DISTRICT606  STATE20       6       782        96         NaN         NaN
210  DISTRICT618  STATE28       5       183        34    656916.0       56.00
348  DISTRICT255  STATE34      10       920        78   1670931.0       73.86
248   DISTRICT63  STATE18       8       230        32    146742.0       82.06
276   DISTRICT57  STATE27       6       121         6    270063.0       85.58
350  DISTRICT254  STATE34      33      3142       286   7102430.0       67.53
252  DISTRICT194   STATE3       8       271        24     78413.0       62.48
10   DISTRICT412   STATE1      11       187        95    476820.0       68.69
478  DISTRICT155  STATE22       5      1116        58   1311008.0       71.11
638  DISTRICT609   STATE6      18      1470       167   2468965.0       74.72
142  DISTRICT570  STATE21       4       399        37   1674714.0       82.20
307    DISTRICT8   STATE4       3       518        41    317618.0       75.51
221  DISTRICT501  STATE28      16      1828       258   4778610.0       65.68
84   DISTRICT326  STATE25       5       256        45    956907.0       80.83
403  DISTRICT529  STATE24      11      1220       134   1271703.0       78.96
254  DISTRICT458   STATE3       3       157        19     82839.0       76.33
271  DISTRICT680  STATE27       5       126        14    196801.0       73.70
256  DISTRICT448   STATE3      10       230        17    112272.0       67.62
247  DISTRICT602  STATE18       7       250        33    136299.0       78.69
291  DISTRICT361  STATE12       3        45        11     83054.0       94.54
338  DISTRICT659  STATE33       6       995       112   1217002.0       79.72
163  DISTRICT366  STATE21       9      1330        86   1579160.0       79.99
524  DISTRICT179   STATE9       4       195        40         NaN         NaN
369  DISTRICT512  STATE23       6       589        30    717169.0       68.35
428  DISTRICT641  STATE31       3       567        61   3343079.0       79.69
365  DISTRICT656  STATE23      19       794        78   1322387.0       62.18
548  DISTRICT283  STATE17       5       765        68   1178973.0       76.04
638  DISTRICT609   STATE6      18      1470       167   2468965.0       74.72
325    DISTRICT1  STATE33       4       534        98    957853.0       69.37
498  DISTRICT443   STATE9      14      1377       282   3116045.0       66.39
426  DISTRICT284  STATE31       4       932        90    822239.0       61.95
669  DISTRICT206  STATE11       3        76        13    105539.0       84.25
621  DISTRICT592   STATE5       3        50        38    816558.0       89.32
368  DISTRICT531  STATE23      16      1063       114   1734005.0       70.48
292  DISTRICT289  STATE12      11       183        51    404054.0       98.50
207  DISTRICT258  STATE21       6       342        53         NaN         NaN
95    DISTRICT90   STATE2       7        29         7    133713.0       89.38
76   DISTRICT282  STATE25       6       201        43   1202811.0       77.46
372  DISTRICT342  STATE23       9      1381       120   1311382.0       57.68
182  DISTRICT670  STATE21      15      1594       142   3478257.0       51.10
145  DISTRICT244  STATE21       9       996        76   1565678.0       73.10
501   DISTRICT54   STATE9       8       684       167   2427346.0       76.60
103  DISTRICT186  STATE13       7      1053       289   2041172.0       67.46
248   DISTRICT63  STATE18       8       230        32    146742.0       82.06
107  DISTRICT630  STATE13       5       825       222   1207293.0       70.14
566   DISTRICT24  STATE17      13      1298       190   3874015.0       82.90
53   DISTRICT126  STATE26       3       197        21    596294.0       68.90
79   DISTRICT447  STATE25       6       284        60    941522.0       69.13
482  DISTRICT119  STATE22       7      1014        54   1240975.0       76.52
595  DISTRICT268  STATE20       7       878       131   1946905.0       76.30

The output show 50 districts that have been selected randomly from the dataset. Note that each has a different literacy rate.

95% Confidence Interval for the Mean District literacy rate base on the above sample data.

In [4]:
sample_mean = sampled_data['OVERALL_LI'].mean()

In [5]:
estimated_standard_error = sampled_data['OVERALL_LI'].std() / np.sqrt(sampled_data.shape[0])

In [6]:
stats.norm.interval(confidence=0.95, loc=sample_mean, scale=estimated_standard_error)

(71.73192867194474, 77.02424154082122)

We have a 95% confidence interval for the mean district literacy rate that stretches from about 71.7% to 77%

95% Cl(71.73, 77.02). Thus the Department of Education will use the estimate of the mean district literacy rate to help make decisions about distribution of funds to different states.

99% Confidence Interval

In [7]:
stats.norm.interval(confidence=0.99, loc=sample_mean, scale=estimated_standard_error)

(70.90044605823485, 77.85572415453112)

There is a 99% confidence interval for the mean district literacy rate that stretches from about 70.9% to 77.9%.
99% Cl:(70.90,77.90)

The relationship between the CL & CI: As the CL gets higher, ther CI gets wider. ie 

With confidence level of 95%, the interval covers 5.27 percentage points (77%- 71.73%)

With a confidence level of 99%, the interval covers 7 percentage points(77.9%-70.9%)

A wider confidence level interval is more likely toinclude the actual population parameter.
With the above results, Department of Education can decide hpw to distribute government resources to improve literacy.